alexandria/core/serializers.py

```diff
@@ -199,27 +199,20 @@ def get_webdav_url(self, instance):
         )
         return instance.get_webdav_url(username, group, host)
 
+    def validate_variant(self, value):
+        # only originals can be created
+        if value and value != models.File.Variant.ORIGINAL:
+            raise serializers.ValidationError(_("Only original files can be created"))
+
+        return models.File.Variant.ORIGINAL
+
     def validate(self, *args, **kwargs):
         """Validate the data.
 
         Validation can be extended by adding ValidatorClass to the
         `validator_classes` attribute of the FileViewSet.
         """
         validated_data = super().validate(*args, **kwargs)
-        if validated_data.get(
-            "variant"
-        ) != models.File.Variant.ORIGINAL and not validated_data.get("original"):
-            file_variant = validated_data.get("variant")
-            raise ValidationError(
-                f'"original" must be set for variant "{file_variant}".'
-            )
-
-        if (
-            variant := validated_data.get("variant")
-        ) == models.File.Variant.ORIGINAL and validated_data.get("original"):
-            raise ValidationError(
-                f'"original" must not be set for variant "{variant}".'
-            )
 
         mime_type = validated_data["content"].content_type
         if mime_type == "application/octet-stream":
```

Comment:
the line below should be removed again `validated_data["mime_type"] = validated_data["content"].content_type`

Reply:
https://github.com/projectcaluma/alexandria/pull/499

Reply:
This is done in another PR: #499